tree/ntuple/doc/Merging.md

```diff
@@ -0,0 +1,75 @@
+    *  **equal or a superset** of the first input (`Union` mode)
+
+Consequences of R3 and R4:
+
+* all columns that were deferred in the first RNTuple remain deferred in the output RNTuple and have the same FirstElementIdx.
+* all projected fields in the input RNTuples remain projected in the output and have the same source field.
+ 
+The following properties are currently true but they are subject to change:
+
+* P1: all output pages have the **same compression** (which may be different from the input pages' compression);
+* P2: the output clusters are **the same as the input clusters**;
+* P3: the output RNTuple **always has 1 cluster group**
+
+## High-level description
+The merging process requires at least 1 input, consisting in a `RPageSource`.
+
+The first input is attached in `EDescriptorDeserializeMode::kForWriting` mode, which doesn't collate the extended header with the non-extended header. Since we use the first input's descriptor as the output schema (barring late model extensions, see later), opening in `kForWriting` mode allows us to write the output to disk preserving the non-extended schema description as per requirement R3. A consequence of this choice is that the merger never produces (new) deferred columns in the output RNTuple's header.
+
+In `Union` mode only, we allow any following input RNTuple to define new fields that don't appear in the first input. These fields, after being validated, are late model extended into the output model and will thus appear in the output RNTuple's extended header on disk. This means that all columns that were not part of the first input's schema become deferred columns in the output RNTuple (unless the first source had 0 entries).
+
+## Descriptor compatibility and validation
+Whenever a new input is processed, we compare its descriptor with the output descriptor to verify that merging is possible.
+
+The comparison function does 3 main things:
+- collect all "extra destination fields" (i.e. fields that exist in the destination but not in this input RNTuple)
+- collect all "extra source fields"
+- collect and validate all common fields.
+
+If the Merging Mode is set to **Filter** we require the "extra destination fields" list to be empty.
+If the Merging Mode is set to **Strict** we require both "extra destination fields" and "extra source fields" to be empty.
+If the Merging Mode is set to **Union**, the "extra source fields" list is used to late model extend the destination model.
+
+As for common fields, they are matched by name and validated as follows:
+- any field that is projected in the destination must be also projected in the source and must be projected to the same field;
+- any field that is not projected in the destination must also not be projected in the source;
+- the field types names must be **identical** (*this could probably be relaxed in the future to allow for different but compatible types*)
```

Comment:
Although not in direct contrast with R1, this sounds related to it, maybe it should be mentioned?